json_schema_fuzz/__init__.py

```diff
@@ -1,15 +1,101 @@
 """JSON schema fuzzer."""
+import json
 import random
 import string
+from decimal import Decimal
 
 import exrex
 
 from .schema_operations import merge
 
+MAX_REJECTED_SAMPLES = 1000
+
+
+class RejectionSamplingFailed(Exception):
+    """
+    Failed to generate sample that satisfies all criteria
+    """
+
+
+def listify(value):
+    """ If value is not a list wrap it in a list """
+    if isinstance(value, list):
+        return value
+    else:
+        return [value]
+
 
 def random_integer(schema):
     """Generate random integer."""
-    return random.randint(0, 10)
+
+    multiple_of = schema.get("multipleOf", Decimal(1))
+
+    default_min = Decimal(1 * multiple_of)
+    default_max = Decimal(100 * multiple_of)
+
+    minimum = max(
+        schema.get("minimum", default_min),
+        schema.get("exclusiveMinimum", default_min) + 1
+    )
+    maximum = min(
+        schema.get("maximum", default_max),
+        schema.get("exclusiveMaximum", default_max) - 1
+    )
+
+    not_multiple_of = schema.get("notMultipleOf", [])
+    if not isinstance(not_multiple_of, list):
+        not_multiple_of = [not_multiple_of]
+
+    for _ in range(MAX_REJECTED_SAMPLES):
+        # Generate new value
+        value = random.randint(minimum, maximum)
+
+        # Verify
+        if value % multiple_of != 0:
```

Comment:
We should be able to pull `multipleOf` stuff out of the rejection sampling. `notMultipleOf` can stay here, though.